src/main/java/it/gov/pagopa/payhub/auth/exception/AuthExceptionHandler.java

```diff
@@ -0,0 +1,47 @@
+package it.gov.pagopa.payhub.auth.exception;
+
+import it.gov.pagopa.payhub.auth.exception.custom.InvalidTokenException;
+import it.gov.pagopa.payhub.auth.exception.custom.TokenExpiredException;
+import it.gov.pagopa.payhub.model.generated.AuthErrorDTO;
+import jakarta.servlet.http.HttpServletRequest;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.core.Ordered;
+import org.springframework.core.annotation.Order;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+
+@RestControllerAdvice
+@Slf4j
+@Order(Ordered.HIGHEST_PRECEDENCE)
+public class AuthExceptionHandler {
+
+    @ExceptionHandler({InvalidTokenException.class, TokenExpiredException.class})
+    @ResponseStatus(HttpStatus.UNAUTHORIZED)
+    public AuthErrorDTO handleInvalidTokenException(ServiceException ex, HttpServletRequest request){
+        logException(ex, request);
+        String message = ex.getMessage();
+
+
+        return new AuthErrorDTO(ex.getCode(), message);
+    }
+
+    public static void logException(ServiceException error, HttpServletRequest request) {
+        if(error.isPrintStackTrace()){
```

Comment:
in which case we want to print a stack trace? `InvalidTokenException` and `TokenExpiredException` are very precise business exceptions which i expect already logs something when they're triggered. In this case what's the added value of a stack trace?
For observability purpose the exception will be visible and inspectable on ApplicationInsight